tests/user.test.js

```diff
@@ -0,0 +1,123 @@
+import chai, { should } from 'chai';
+import chaiHttp from 'chai-http';
+
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+import app from '../src';
+
+import { signup, signin } from '../src/controllers/userController';
+import * as userService from '../src/services/userAuth';
+import database from '../src/models/index';
+
+import user from './mock';
+
+chai.use(chaiHttp);
+chai.use(sinonChai);
+should();
+
+const users = database.User;
+
+describe('signup Route', () => {
+  it('should post to /api/v1/auth/signup', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/signup').send(user);
+
+    response.should.have.status(200);
+    response.body.status.should.eql(200);
+    response.body.should.have.property('token');
+    response.body.should.have.property('user');
+  });
+
+  it('should return an error if user is already registered', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/signup').send(user);
+
+    response.should.have.status(409);
+    response.body.status.should.eql('error');
+    response.body.should.have.property('error');
+  });
+});
+
+describe('signin Route', () => {
+  after(async () => {
+    await users.destroy({
+      where: {},
+      truncate: true
+    });
+  });
+
+  it('should throw an error for wrong password', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/signin').send({
+      email: user.email,
+      password: 'wrong password'
+    });
+
+    response.should.have.status(400);
+    response.body.status.should.eql('error');
+    response.body.error.message.should.equal('invalid user name or password');
+  });
+
+  it('should throw an error for wrong email', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/signin').send({
+      email: 'wrongemail@gmail.com',
+      password: user.password
+    });
+
+    response.should.have.status(400);
+    response.body.status.should.eql('error');
+    response.body.error.message.should.equal('you are not yet registered');
+  });
+
+  it('should post to /api/v1/auth/signin', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/signin').send({
+      email: user.email,
+      password: user.password
+    });
+
+    response.should.have.status(200);
+    response.body.status.should.eql(200);
+    response.body.should.have.property('token');
```

Comment:
I think this test is a bit shallow. Can you explicitly test what you are the token to be e.g `number`, `string`  and can you also test that what is been returned within the `user` property is what is expected